### PR TITLE
 Vpnaas: Update IKE policy

### PR DESCRIPTION
**Pull request by simonre**
Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

For #723 

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/openstack/neutron-vpnaas/blob/058469e1b99b647537a5228c6a384d93df5484df/neutron_vpnaas/db/vpn/vpn_db.py#L338

### CODE REVIEW

**Comment by simonre** [comment]
@jtopjian This is ready for review

File: openstack/networking/v2/extensions/vpnaas/ikepolicies/requests.go [hunk 1]
```diff
@@ -105,3 +105,42 @@ func Create(c *gophercloud.ServiceClient, opts CreateOptsBuilder) (r CreateResul
...
+	ToPolicyUpdateMap() (map[string]interface{}, error)
+}
+
+type LifetimeUpdateOpts struct {
+	Units Unit `json:"units,omitempty"`
+	Value int  `json:"value,omitempty"`
+}
+
+// UpdateOpts contains the values used when updating an IKE policy
+type UpdateOpts struct {
```
Comment (jtopjian) [inline openstack/networking/v2/extensions/vpnaas/ikepolicies/requests.go]: It looks like `Phase1Negotiation` can be updated: https://github.com/openstack/neutron-vpnaas/blob/master/neutron_vpnaas/extensions/vpnaas.py#L418

**Comment by simonre** [comment]
@jtopjian I added the field and I also noticed that you can update the IKE version as well so I added that, too. 

In cases like this where it's just a one or two line change, would you mind if I just rebased without you explicitly asking for it? 
Also: Yes, both this and the "Update IPSec Policy" branch can be merged now. The problem with the "Update service" branch is that `Service` has a field `Status` which has to be a certain value (`ACTIVE`) for openstack to allow an update of its values. This field however does not change from `PENDING_CREATE` to `ACTIVE` until an IPSec Site Connection for the VPN service has been created. IPSec and IKE policies do not have this problem and can be updated immediately upon creation.

**Comment by jtopjian** [comment]
@simonre Rebasing in certain cases (like this) is fine, so you're more than welcome to do it in preparation.

The house rule about not rebasing or amending commits is for someone to make a dedicated commit of the requested changes so the reviewer can distinguish those changes from the original.

**Comment by jtopjian** [comment]
@simonre Looks like this one didn't make this round of merges before requiring another rebase :(

**Comment by simonre** [comment]
@jtopjian I think this is ready now.

**Pull request merged**